kDriveCore/Data/UploadQueue/Servicies/PhotoLibraryUploader+Scan.swift

```diff
@@ -68,24 +69,16 @@ public extension PhotoLibraryUploader {
 
     // MARK: - Private
 
-    private func updateLastSyncDate(_ date: Date, using realm: Realm = DriveFileManager.constants.uploadsRealm) {
-        if let settings = realm.objects(PhotoSyncSettings.self).first {
-            try? realm.safeWrite {
-                if !settings.isInvalidated {
-                    settings.lastSync = date
-                }
-            }
-            if settings.isInvalidated {
-                _settings = nil
-            } else {
-                _settings = PhotoSyncSettings(value: settings)
-            }
+    private func updateLastSyncDate(_ date: Date, writableRealm: Realm) {
+        if let settings = writableRealm.objects(PhotoSyncSettings.self).first,
+           !settings.isInvalidated {
+            settings.lastSync = date
         }
     }
 
     private func addImageAssetsToUploadQueue(assetsFetchResult: PHFetchResult<PHAsset>,
```

Comment:
I reworked this part to seamlessly use Transactionable, while still be able to incrementally write to DB.
I did some manual testing, works.